USING_WITH_PYRIGHT.md

```diff
@@ -0,0 +1,305 @@
+# Pyright and Pylance Together
+
+[Pyright](https://github.com/Microsoft/pyright) is an open source version of a Python type checker that forms the basis for Pylance. Both Pyright and Pylance can be used inside of an IDE to provide a [language server](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide#why-language-server) for Python code.
+
+Pyright can also be used on the [command line](https://github.com/microsoft/pyright/blob/main/docs/command-line.md) to produce json or text output. This can be used in a variety of situations, like a pre-commit hook, or a [github action](https://github.com/jakebailey/pyright-action).
+
+However, Pylance and Pyright don't always produce the same results. The reason for this is a difference in configuration. 
+
+## Stub differences
+
+Pylance comes bundled with a bunch of stubs, usually found here in the installation:
+
+```
+<pylance-extension-install>/dist/bundled
+<pylance-extension-install>/dist/native-stubs
+```
+
+In order for Pyright to produce the same errors as Pylance it need to use the same stub files. 
+
+This might be done by copying the contents of each of those folders to a new folder. 
+
+You'd then reference that new folder in your [pyrightconfig.json](https://microsoft.github.io/pyright/#/configuration)
+
+```json
+{
+    "stubPath": "./pylanceStubs"
+}
+```
+
+Where `pylanceStubs` contains the contents of the `bundled` and `native-stubs` folders.
+
+We have plans to make this easier in the [future](https://github.com/microsoft/pylance-release/discussions/3638).
+
+## Settings differences
+
+Both Pylance and Pyright have [settings](https://github.com/microsoft/pylance-release#settings-and-customization) to control behavior. For the most part, they use the same values, but the few differences do have an impact on the analysis.
+
+| Setting | Pylance default | Pyright default | Description | Potential Impact |
+|----|----|----|----|----|
+| python.analysis.autoSearchPaths | true | false|  Adds 'src' to the list of search paths | This may change what files are found when analyzing. So if you're getting missing imports for things in your 'src' tree, this might be why |

```

Comment:
I think you can delete the section on "python.analysis.autoSearchPaths". I'm pretty sure that pyright and pylance use the same default value here, so if you don't override the default in your language server settings, you will get the same behaviors.

Reply:
This looks like the default is 'false' for pyright command line?
https://github.com/microsoft/pyright/blob/09c79ed3e284b953b4d96c240a2dca91d74b2eee/packages/pyright-internal/src/analyzer/service.ts#L699

Maybe I'm not following the code flow correctly here.

Reply:
I guess I was mistaken. Looks like this default is different too. Another one we should consider changing to make them the same. I'm fine switching the default to true for pyright if you don't see any problem with that.

Reply:
Well except that people using it will suddenly get subtle differences. I guess we have no good way to notify people of these sort of things other than release notes.

Reply:
Yes, in theory a user could see a change in behavior, but I think it's unlikely to affect many (any?) pyright users. If a user has a "src" directory, they probably already configure it as an import search path using "extraPaths".